servers/cromwell/jobs/controllers/jobs_controller.py

```diff
@@ -128,29 +129,90 @@ def get_job(id, **kwargs):
 
 
 def format_task(task_name, task_metadata):
+    # check to see if task is scattered
+    if task_metadata[-1].get('shardIndex') != -1:
+        return format_scattered_task(task_name, task_metadata)
+    latest_attempt = task_metadata[-1]
     return TaskMetadata(
         name=remove_workflow_name(task_name),
-        execution_id=task_metadata.get('jobId'),
+        execution_id=latest_attempt.get('jobId'),
         execution_status=task_statuses.cromwell_execution_to_api(
-            task_metadata.get('executionStatus')),
-        start=_parse_datetime(task_metadata.get('start')),
-        end=_parse_datetime(task_metadata.get('end')),
-        stderr=task_metadata.get('stderr'),
-        stdout=task_metadata.get('stdout'),
-        inputs=update_key_names(task_metadata.get('inputs', {})),
-        return_code=task_metadata.get('returnCode'),
-        attempts=task_metadata.get('attempt'),
-        call_root=task_metadata.get('callRoot'),
-        job_id=task_metadata.get('subWorkflowId'))
+            latest_attempt.get('executionStatus')),
+        start=_parse_datetime(latest_attempt.get('start')),
+        end=_parse_datetime(latest_attempt.get('end')),
+        stderr=latest_attempt.get('stderr'),
+        stdout=latest_attempt.get('stdout'),
+        inputs=update_key_names(latest_attempt.get('inputs', {})),
+        return_code=latest_attempt.get('returnCode'),
+        attempts=latest_attempt.get('attempt'),
+        call_root=latest_attempt.get('callRoot'),
+        job_id=latest_attempt.get('subWorkflowId'),
+        shard_statuses=None)
+
+
+def format_scattered_task(task_name, task_metadata):
+    temp_status_collection = {}
+    current_shard = ''
+    minStart = _parse_datetime(task_metadata[0].get('start'))
+    maxEnd = _parse_datetime(task_metadata[0].get('end'))
+
+    # go through calls in reverse to grab the latest attempt if there are multiple
+    # get earliest start time and latest end time
+    for shard in task_metadata[::-1]:
+        if current_shard != shard.get('shardIndex'):
+            status = task_statuses.cromwell_execution_to_api(
+                shard.get('executionStatus'))
+            if status not in temp_status_collection:
+                temp_status_collection[status] = 1
+            else:
+                temp_status_collection[
+                    status] = temp_status_collection[status] + 1
+        if minStart > _parse_datetime(shard.get('start')):
+            minStart = _parse_datetime(shard.get('start'))
+        if shard.get('executionStatus') not in ['Failed', 'Done']:
+            maxEnd = None
+        if maxEnd is not None and maxEnd < _parse_datetime(shard.get('end')):
```

Comment:
What happens if `_parse_datetime(shard.get('end'))` is `None`? Is it possible?

Reply:
`_parse_datetime(None)` will return `None` (see line 334)